File: src/main/java/com/example/springpractice/domain/Member.java
package com.example.springpractice.domain;

import com.example.springpractice.domain.vo.Address;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member extends BaseTimeEntity {

    @Id
    @GeneratedValue
    private Long id;
    private String name;
    private String nickname;
    @Embedded
    private Address address;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @Builder
    public Member(Long id, String name, String nickname, Address address, Team team) {
        this.id = id;
        this.name = name;
        this.nickname = nickname;
        this.address = address;
        this.team = team;
    }
}

File: spring-batch-kotlin/src/main/kotlin/com/example/springbatchkotlin/domain/Member.kt
package com.example.springbatchkotlin.domain

import jakarta.persistence.*
import org.hibernate.envers.Audited

@Table(name = "members")
@Entity
@Audited
class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,
    val name: String,
    val loginId: String
) {

    companion object {
        fun of(name: String, loginId: String): Member {
            return Member(id = null, name = name, loginId = loginId)
        }
    }
}
